server/controllers/bookingRequest.js

```diff
@@ -0,0 +1,81 @@
+const asyncHandler = require("express-async-handler");
+const User = require("../models/User");
+const BookingRequest = require("../models/BookingRequest");
+
+// @route POST /booking-requests/
+// @desc Create a new request
+// @access Private
+
+exports.createBookingRequest = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+  const user = await User.findById(userId);
+  const { sitterId, start, end, description } = req.body;
+
+  const sitter = await User.findById(sitterId);
+  if (sitter) {
+    const bookingRequest = await BookingRequest.create({
+      userId: user,
+      sitterId: sitter,
+      start,
+      end,
+      description,
+    });
+
+    res.status(201).json({
+      success: {
+        bookingRequest: {
+          requestId: bookingRequest._id,
+          userId: bookingRequest.userId._id,
+          sitterId: bookingRequest.sitterId._id,
+          start: bookingRequest.start,
+          end: bookingRequest.end,
+          description: bookingRequest.description,
+        },
+      },
+    });
+  } else {
+    res.status(500);
+    throw new Error("Unable to find the dog sitter info");
+  }
+});
+
+// @route PATCH /booking-requests/:requestId
+// @desc Update a booking request (only for dog owners)
+// @access Private
+
+exports.updateBookingRequest = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+  const { requestId } = req.params;
+  const fieldsToUpdate = ({ start, end, description } = req.body);
```

Comment:
```suggestion
  const fieldsToUpdate = ({ start, end, status, description } = req.body);
```
The owner of a request should also be able to update the status of a request (accept/decline it)